### PR TITLE
Update debug/dump cmd with Cluster-Environment-Name header

### PR DESCRIPTION
**Pull request by ryanhall07**
**What this PR does / why we need it**:
This is needed since the dump includes the PlacmentSource

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
NONE

**Does this PR require updating code package or user-facing documentation?**:
NONE
```

### CODE REVIEW

File: docs/troubleshooting/index.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ This endpoint can be used on both the db nodes as well as the coordinator/query
 To use this, simply run the following on either the M3DB debug listen port or the regular port on M3Coordinator.
```
Comment (schallert) [inline docs/troubleshooting/index.md:41]: Can you add something along the lines of "if the placement you are interested in is under a separate `env` than the service you are downloading the dump from, you may need to set the `Cluster-Environment-Name` header"

Comment (schallert) [reply docs/troubleshooting/index.md:41]: Never mind! The fallback behavior isn't consistent.

**Review by schallert — APPROVED** [review]

File: docs/troubleshooting/index.md [hunk 2]
```diff
@@ -40,8 +40,14 @@ This endpoint can be used on both the db nodes as well as the coordinator/query
 
 To use this, simply run the following on either the M3DB debug listen port or the regular port on M3Coordinator.
 
+You may need to include the following headers:
+
+--8<--
```
Comment (nbroyles) [inline docs/troubleshooting/index.md:45]: What's this mean? Renders weirdly when I click the markdown view.

Comment (ryanhall07) [reply docs/troubleshooting/index.md:45]: yea it's as mkdocs thing, that will embed other pages. we do something similar for https://m3db.github.io/m3/operational_guide/placement_configuration/

see https://raw.githubusercontent.com/m3db/m3/master/docs/operational_guide/placement_configuration.md

**Pull request merged**